### PR TITLE
Fix automatic generation of spatial function types files

### PR DESCRIPTION
**Pull request by craigtaverner**
The automatic mapping of spatial function names from class names was not working for spatial types, so the automatic generation of these files did not happen, and in fact existing files were deleted when running the full test suite (this only happened infrequently, hence the lack of discovery of this issue for so long).

Automatic filename generation did the following: `ToGeoPointTests` --> `to_geo_point`, while the desired function name is `to_geopoint`. We fixed this by supporting an annotation on the test class:

```
@FunctionName("to_geopoint")
public class ToGeoPointTests extends AbstractFunctionTestCase {
```

This was applied to four test classes for which default names were being incorrectly deduced.

In addition, the generation of aggregation functions types does not yet exist at all, so the `st_centroid.asciidoc` file was always deleted. Until such support exists, this files contents will be moved back into the function definition file. As an option we could delete this content, since no other aggregation functions bother to explain their input types in the docs. Personally I think the information is useful, especially for when we expand support of this function for `geo_shape` and `cartesian_shape`.

The railroad diagrams for syntax are now also created, however, not all functions in the documentation actually use these, and certainly none of the `TO_*` type-casting functions do, so we'll not include links to them from the docs, and leave that to the docs team to decide. Personally, while these diagrams are pretty, they contain no additional informational content, and in fact give a cluttered impression to the documentation's visual appeal.

### CODE REVIEW

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java [hunk 1]
```diff
@@ -1116,12 +1116,23 @@ private static void renderTypesTable(List<String> argNames) throws IOException {
...
     }
 
     private static String functionName() {
-        return StringUtils.camelCaseToUnderscore(getTestClass().getSimpleName().replace("Tests", "")).toLowerCase(Locale.ROOT);
+        return mapFunctionName(
+            StringUtils.camelCaseToUnderscore(getTestClass().getSimpleName().replace("Tests", "")).toLowerCase(Locale.ROOT)
+        );
+    }
+
+    private static String mapFunctionName(String name) {
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java:1129]: This is the simplest change to achieve a fix, but presumably there could be a more generic change with some annotation to the test class, or test rule, or reflection lookup of a class method. For example, perhaps classes that wish to override this simply define `MyTests::functionName` and the existing static functionName method first tries to get that before falling back to automatic name mapping.

**Comment by leemthompo** [comment]
> Personally, while these diagrams are pretty, they contain no additional informational content, and in fact give a cluttered impression to the documentation's visual appeal.

I tend to agree.

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java:1129]: I do like the idea of annotating something - though it looks like it'd have to be the test class.

**Comment by nik9000** [comment]
> I tend to agree.

If we want to drop them I'm fine with it. I added them because I thought someone asked for them. But they are kind of a thing to maintain. OTOH, we still have to maintain the table.

**Comment by nik9000** [comment]
> Automatic filename generation did the following: `ToGeoPointTests` --> `to_geo_point`, while the desired function name is `to_geopoint`. We fixed this by adding function name exceptions for `point` and `shape`.

Should we call it `ToGeopoint` in that case? It's not that I love the heuristics in the test but they are telling us we're doing something inconsistent.

OTOH, I like the idea of a little annotation to pick this up.

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java (hunk 1 above)
Comment (craigtaverner) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java]: OK. I've changed it to use an annotation. That is certainly more generic.

**Review by nik9000 — APPROVED** [review]

**Pull request merged**